### ISSUE TITLE
i don't think it is right

### ISSUE THREAD

**Issue by tengshaofeng** [v-nehzil/focal_Loss#2]
focalLoss = -tf.reduce_sum(y * ((1 - probs) ** gama) * tf.log(probs))

when one elem in y is 0, then loss is 0?

**Comment by lengly** [comment]
Your point is "what is cross entropy"
For binary classification, it's y * -log(p) + (1-y) * -log(1-p). 
But for multi-class classification, it's y * -log(p)

**Comment by tengshaofeng** [comment]
how about this?

def index_along_every_row(array, index):
    N, _ = array.shape
    return array[np.arange(N), index]

def focal_loss(features, labels, alpha=0.25, g=2):
    prob = tf.nn.softmax(features)  # # (features + 1e-7)
    pt = tf.py_func(index_along_every_row, [prob, labels], [tf.float32])[0]
    cross_entropy = tf.log(tf.clip_by_value(pt, 1e-7, 1.0))
    weight = tf.pow(tf.subtract(1.0, pt), g)
    loss = tf.reduce_mean(-alpha*tf.multiply(weight, cross_entropy))
    return loss
